# Mining the Crowd  
### Collective Intelligence Signals for Diagnosis and Design in Community Notes

**Course:** MACS 40123  
**Author:** Jonathan Monroe  
**Date:** October 2025

Digital platforms now make decisions that mirror methods of governance. They decide which misinformation claims gain visibility, which corrections are surfaced in response, and which moderating voices are treated as credible. Those decisions happen fast, and mostly without public justification.

Using X Community Notes as a case study, our work maps the structural accountability of a human-led moderation system—who participates, how agreement forms, and how judgments become platform outcomes—and uses collective intelligence principles to evaluate that structure and identify where it could be improved.

---

### Clarifying “Structural Accountability”

Platforms now routinely present systems like Community Notes as “the crowd” speaking. That claim is politically powerful: it lets the platform frame moderation and misinformation labeling as democratic, neutral, and scalable. If the underlying structure is not actually diverse, independent, decentralized, or well-aggregated, then that framing is illegitimate.

I use the term "structural accountability" to describe the degree to which the system’s internal logic satisfies the conditions for collective intelligence: who is allowed to contribute judgment, how consensus emerges among contributors, and how that consensus is converted into platform outcomes.

In this project, I map that structure directly. I identify who is participating in judgment, how agreement forms, and how those judgments are translated into final platform decisions. Through large-scale data mining, I measure those patterns, critique them, and use that critique to surface where the system could be redesigned.

I treat collective intelligence principles—diversity, independence, decentralization, and aggregation quality—as diagnostics of that structure. Each principle is a measurement axis. It tells us how judgment is organized, whose judgment ends up mattering, and how that judgment becomes something the platform treats as authoritative.

By quantifying structural accountability, this project makes the Community Notes system’s legitimacy claim testable. If parts of the system satisfy these collective intelligence conditions, that shows where the platform can defend its governance structure against accusations of systematic bias. If they don’t, that reveals where the Community Notes decision pipeline is unreliable as a misinformation detection mechanism—and, by extension, where its design needs to change.

---

### Research Question

> **Using X Community Notes as a case study, how can collective intelligence signals be used to assess the structural accountability of the platform's decision pipeline?**

---

### Notebook Overview
This notebook is the exploratory analysis and project proposal for our large-scale data mining study of the X Community Notes system.

This notebook does three things:

1. States the research question and why it matters.  
2. Describes the data and shows initial exploratory analysis.  
3. Lays out the methodological plan to mine “collective intelligence signals” that can inform platform design.

### Background and Related Work

Collective intelligence research makes a basic but extremely important claim: “the crowd” is only useful under specific structural conditions. Large groups outperform isolated experts when: individuals bring different information and perspectives rather than repeating the same narrative (diversity); people make their judgments independently instead of copying those who spoke first (independence); influence and information are distributed and locally grounded rather than centralized in a small bloc (decentralization); there is a defined way to turn many individual judgments into a single actionable outcome (aggregation). When those conditions are met, aggregated group judgments can be more accurate, more stable, and in some case, more robust to partisan spin than judgments from professional elites (Surowiecki 2005; Lorenz et al. 2011; Almaatouq et al. 2020). They are not met, for example, when early ratings anchor on later ones, or when a small cluster of high-activity participants dominate decisions. The “crowd” can underperform, and even amplify error, as what appears to be consensus could, in reality, be imitation (Lorenz et al. 2011; Surowiecki 2005).

These structural conditions map directly onto the governance problem for misinformation. Platforms now frame certain moderation and correction mechanisms as if they are expressions of the public will. X’s Community Notes system, for example, is explicitly described as “the community providing context,” rather than the platform declaring a post misleading. That framing matters politically, because it recasts content labeling and the assignment of credibility as democratic, neutral, and scalable, rather than top-down editorial enforcement (Gillespie 2018; Matias 2019). But that claim of legitimacy only holds if the underlying process actually behaves like a collective intelligence system— whereby judgment is meaningfully distributed, agreement emerges without herding, and the final “helpful / not helpful” outcome reflects a consistent aggregation of many distinct inputs rather than the preferences of a small gatekeeping layer (Surowiecki 2005; Lorenz et al. 2011; Almaatouq et al. 2020).

Recent empirical work has started to test those claims directly in the context of crowdsourced fact-checking and Community Notes. Studies of platform-scale fact-checking crowds find that large, politically mixed groups of raters can collectively identify misinformation in ways that approximate expert judgments, and can do so at a speed and scale that expert review alone cannot match (Martel et al. 2024). Research on Community Notes in particular has shown two things: first, that attaching a crowdsourced note to a post reduces downstream engagement and resharing of misleading content, suggesting that these notes can suppress diffusion of misinformation (Chuai et al. 2024); and second, that only a subset of misleading posts ever receives such a note, often after most of the spread has already occurred, which limits system-wide impact (Pilarski et al. 2024). Related work has also begun to analyze how Community Notes surfaces notes that draw agreement from ideologically distant raters, suggesting that the platform is not just counting votes but actively privileging “bridging consensus” across political sub-communities as a credibility signal (Kim et al. 2025; Li et al. 2025). This is an explicit form of aggregation logic: the platform is operationalizing not just “how many people agreed,” but “who agreed with whom,” which is an institutional design choice, not a neutral fact.

This project builds on that line of work by treating Community Notes as a decision pipeline—a system that turns individual human judgments into platform-visible outcomes. Instead of asking abstractly whether misinformation is being “handled,” I evaluate what I call structural accountability: the extent to which the internal logic of the system satisfies the core conditions identified in collective intelligence research. Concretely, I measure four diagnostic dimensions that correspond to those conditions. (1) Diversity: whether attention is distributed across different kinds of claims and topics, or concentrated in a narrow band of high-conflict narratives. (2) Independence: whether later raters appear to be making their own calls or simply reproducing early ratings, which would indicate herding pressure. (3) Decentralization: whether judgment is broadly shared across many contributors or effectively concentrated in a small, high-volume stratum that acts as a de facto moderation layer. (4) Aggregation quality: whether the platform’s final status decisions on notes (“helpful,” “needs more ratings,” etc.) track the underlying pattern of ratings in a stable, monotonic, and explainable way, rather than being arbitrary or selectively enforced. Each of these diagnostic axes corresponds to a known requirement for reliable crowd decision-making (Surowiecki 2005; Lorenz et al. 2011; Almaatouq et al. 2020) and to emerging empirical work on crowdsourced misinformation correction at platform scale (Martel et al. 2024; Drolsbach et al. 2024).

The result is that “the crowd decided” stops being a marketing claim and becomes an empirical question with defined failure modes. If the structure behind Community Notes satisfies these collective intelligence conditions, the platform can plausibly defend its governance process against accusations that it is partisan or arbitrary. If it does not, then decisions presented as “crowd-sourced” are in practice concentrated, imitative, or weakly aggregated—and the system is not behaving as a trustworthy misinformation filter at scale (Gillespie 2018; Matias 2019; Drolsbach et al. 2024).


## Exploratory Data Overview

Before applying large-scale data mining methods, this section examines the structure and scale of the Community Notes datasets to establish what forms of analysis are feasible and what the data can reveal about collective judgment on the platform.

The data span 2021–2025 and capture how notes are written, rated, and resolved. Each table serves a specific role in the decision pipeline:

- **`notes.parquet`** – one record per note, containing metadata about the content, author, and time of creation.  
- **`ratings.parquet`** – one record per rater–note interaction, indicating how a participant judged the note’s helpfulness and why.  
- **`noteStatusHistory.parquet`** – timestamped logs of each note’s evaluation state (e.g., *NEEDS_MORE_RATINGS*, *CURRENTLY_RATED_HELPFUL*).  
- **`userEnrollment.parquet`** – background information on rater eligibility and enrollment cohort.

Together, these tables allow reconstruction of how human judgments flow through the Community Notes pipeline—from initial contribution, to crowd evaluation, to platform-level decision.  

Below, we verify dataset scale, record distribution, and basic temporal coverage to establish the foundation for later analyses of diversity, independence, decentralization, and aggregation quality.


In [2]:
from pathlib import Path
from pyspark.sql import SparkSession

spark = SparkSession.builder.getOrCreate()

ProjectRoot = Path.cwd().parent
DataDir = ProjectRoot / "data" / "interim"

RatingsPath = DataDir / "ratings_parquet" / "ratings.parquet"
NotesPath = DataDir / "processed" / "notes.parquet"
StatusPath = DataDir / "processed" / "noteStatusHistory.parquet"
EnrollPath = DataDir / "processed" / "userEnrollment.parquet"

ratings = spark.read.parquet(str(RatingsPath))
notes = spark.read.parquet(str(NotesPath))
status = spark.read.parquet(str(StatusPath))
enroll = spark.read.parquet(str(EnrollPath))

print(f"Ratings count: {ratings.count():,}")
print(f"Notes count: {notes.count():,}")
print(f"Status records: {status.count():,}")
print(f"Enrollment records: {enroll.count():,}")

ratings.printSchema()


Ratings count: 171,721,201
Notes count: 2,119,356
Status records: 2,284,533
Enrollment records: 1,286,325
root
 |-- noteId: long (nullable = true)
 |-- raterParticipantId: string (nullable = true)
 |-- createdAtMillis: long (nullable = true)
 |-- version: long (nullable = true)
 |-- agree: long (nullable = true)
 |-- disagree: long (nullable = true)
 |-- helpful: long (nullable = true)
 |-- notHelpful: long (nullable = true)
 |-- helpfulnessLevel: string (nullable = true)
 |-- helpfulOther: long (nullable = true)
 |-- helpfulInformative: long (nullable = true)
 |-- helpfulClear: long (nullable = true)
 |-- helpfulEmpathetic: long (nullable = true)
 |-- helpfulGoodSources: long (nullable = true)
 |-- helpfulUniqueContext: long (nullable = true)
 |-- helpfulAddressesClaim: long (nullable = true)
 |-- helpfulImportantContext: long (nullable = true)
 |-- helpfulUnbiasedLanguage: long (nullable = true)
 |-- notHelpfulOther: long (nullable = true)
 |-- notHelpfulIncorrect: long (nullable = t

In [3]:
summary = {
    "Ratings Rows": ratings.count(),
    "Distinct Raters": ratings.select("raterParticipantId").distinct().count(),
    "Distinct Notes Rated": ratings.select("noteId").distinct().count(),
    "Notes Total": notes.count(),
    "Status Entries": status.count(),
    "User Enrollments": enroll.count()
}

for k, v in summary.items():
    print(f"{k:<25}: {v:,}")

25/10/24 22:59:34 WARN RowBasedKeyValueBatch: Calling spill() on RowBasedKeyValueBatch. Will not spill but return 0.
25/10/24 22:59:34 WARN RowBasedKeyValueBatch: Calling spill() on RowBasedKeyValueBatch. Will not spill but return 0.
25/10/24 22:59:34 WARN RowBasedKeyValueBatch: Calling spill() on RowBasedKeyValueBatch. Will not spill but return 0.
25/10/24 22:59:34 WARN RowBasedKeyValueBatch: Calling spill() on RowBasedKeyValueBatch. Will not spill but return 0.
25/10/24 22:59:34 WARN RowBasedKeyValueBatch: Calling spill() on RowBasedKeyValueBatch. Will not spill but return 0.
25/10/24 22:59:34 WARN RowBasedKeyValueBatch: Calling spill() on RowBasedKeyValueBatch. Will not spill but return 0.
25/10/24 22:59:34 WARN RowBasedKeyValueBatch: Calling spill() on RowBasedKeyValueBatch. Will not spill but return 0.
25/10/24 22:59:34 WARN RowBasedKeyValueBatch: Calling spill() on RowBasedKeyValueBatch. Will not spill but return 0.
25/10/24 22:59:35 WARN RowBasedKeyValueBatch: Calling spill() on

Ratings Rows             : 171,721,201
Distinct Raters          : 1,154,092
Distinct Notes Rated     : 2,137,599
Notes Total              : 2,119,356
Status Entries           : 2,284,533
User Enrollments         : 1,286,325


### Scale and Coverage

The Community Notes dataset captures a complete snapshot of the crowd-moderation system at platform scale. It includes:

| Dataset | Description | Count |
|----------|--------------|-------|
| **Ratings** | Individual judgments linking raters to notes | 171,721,201 |
| **Distinct Raters** | Unique participants who have rated at least one note | 1,154,092 |
| **Distinct Notes Rated** | Notes that have received at least one rating | 2,137,599 |
| **Notes Total** | All created notes, whether rated or not | 2,119,356 |
| **Status History Entries** | Timestamped evaluation logs for note status transitions | 2,284,533 |
| **User Enrollments** | Distinct participants enrolled in the system | 1,286,325 |

The dataset therefore represents more than **170 million human judgments** across over **2 million distinct notes**, rated by more than **a million contributors**. The slight discrepancy between *total notes* and *distinct notes rated* reflects that some notes were re-rated or reissued under different versions following content edits or moderation feedback loops.

This scale enables quantitative tests of structural accountability: how participation, agreement, and aggregation are distributed across millions of decisions. The magnitude of activity ensures that statistical regularities—such as skewed participation, clustering of topics, or correlation of early and late ratings—can be measured with precision rather than inferred from small samples.


## Methodological Design

This project mines what I call "collective intelligence signals"—statistical traces of how a crowd produces, evaluates, and converges on shared judgments. Each diagnostic corresponds to one of the classical conditions for collective intelligence—diversity, independence, decentralization, and aggregation—and each is operationalized using the large-scale Community Notes datasets described above. Together, they allow me to quantify what I term "structural accountability": whether the system’s internal logic satisfies the conditions under which crowds make sound, unbiased collective decisions.

The methodological design, therefore, moves from descriptive mapping to diagnostic measurement, applying established data mining techniques to unveil those structural signals from behavioral data.

---

#### Diversity 

To evaluate diversity, I cluster note embeddings to estimate the topical structure of crowd attention. Each note’s text embedding (generated from the “summary” field using all-MiniLM-L6-v2) is projected into a lower-dimensional space via PCA and UMAP, and then clustered using K-Means. The resulting topics capture high-level domains—such as political fact-checks, media hoaxes, or health misinformation—that define the semantic spread of the system’s focus.

This procedure was implemented in the soon-to-be-uploaded topic_clustering.ipynb. Topic prevalence, entropy, and coverage distributions are computed to measure whether attention is dispersed across multiple informational domains or concentrated in a small set of high-salience topics. Topic clusters are also mapped against note helpfulness rates, revealing whether certain subjects dominate collective attention or produce systematic agreement/disagreement patterns.

---

#### Independence

Independence tests whether raters form their own judgments or reproduce those of others. Using the full rating history, I measure rater–rater dependence via pairwise similarity of rating vectors (as implemented in the soon-to-be-uploaded dependence_modeling.ipynb). Locality-sensitive hashing (LSH) and Jaccard similarity sampling were initially used to estimate dependence density across the population, yielding the structural similarity rate (≈3.4%), indicating high independence overall.

For the purpose of this project, I will also estimate independence in a second way by comparing early vs. late rating agreement on each note and computing the correlation between early helpfulness proportions and final outcomes. If later raters consistently replicate earlier votes, independence declines, signaling herding or visibility bias. Stability across sample fractions and temporal intervals provides a robustness check on these findings.

---

#### Decentralization

Decentralization will be measured as a bipartite network between raters and notes. This structure makes visible how judgment flows through the platform—whether it’s dispersed across many contributors or concentrated in a small, recurring subset. Each rater’s degree reflects how many notes they have rated, and each note’s degree shows how many unique raters engaged with it. From these distributions, I compute inequality metrics such as the Gini coefficient and degree entropy to capture whether participation is broad or bottlenecked.

I will then project this bipartite graph onto the rater layer to analyze patterns of co-participation: how often the same raters appear together across notes. Network modularity and connected-component analysis will reveal whether these contributors form localized clusters or remain broadly integrated. If most rating activity clusters around a small, densely connected group, the system is less decentralized than it appears; if connections are widespread and weakly clustered, that indicates a genuinely distributed decision structure.

---

#### Aggregation

Aggregation test whether Community Notes consistently translates many individual judgments into coherent platform decisions. I will model the mapping between note-level rating distributions and final status outcomes (“Helpful,” “Needs More Ratings,” etc.) to evaluate whether those outcomes follow a stable, monotonic relationship with the underlying helpful vote share. Logistic and ordinal regression models will estimate how well collective outcomes can be predicted directly from crowd behavior.

I will also assess whether aggregation behaves uniformly across topics. If some domains systematically achieve “Helpful” designations under weaker consensus, that would indicate that aggregation thresholds are uneven and encode topical bias. To identify these asymmetries, I will compute a topic-level “overperformance” metric—comparing each topic’s helpful rate to the global average—as an indirect indicator of structural preference.

To benchmark aggregation quality, I will use link analysis to see when Community Notes and professional fact-checkers are talking about the same claim. I will extract every outbound URL that appears in Community Notes (for example, links to Snopes, PolitiFact, Reuters Fact Check, AFP Fact Check, etc.). I will treat those linked fact-check articles as an external reference set.

If a note that earns a “Helpful” status links to a third-party fact-check article that explicitly disputes the same claim, that’s evidence that the platform’s aggregation process elevated a correction that is externally verifiable. If a note fails to become “Helpful” despite linking to an established fact-check, or if a note systematically disagrees with a third-party fact check, that’s evidence of drift between what the crowd judges as “helpful” and what professional fact-checkers have already documented.

## Feasibility

All of the analyses proposed here are technically and computationally feasible. The full Community Notes corpus—spanning roughly 170 million ratings, 2.1 million notes, and 1.1 million raters—has already been ingested, converted to Parquet, and validated through earlier preprocessing notebooks. During preprocessing, I filtered out non-English notes, standardized timestamps, reconciled mismatched note and rater identifiers across tables, de-duplicated entries, and cleaned summary text fields for embedding generation. The diagnostic procedures build directly on Spark pipelines for embedding, clustering, and network modeling, each of which has already been successfully run on Midway3’s caslake and bigmem partitions. Intermediate datasets (which are still being organized for reproducability) make each stage modular and reproducible. This architecture ensures that every diagnostic—diversity, independence, decentralization, and aggregation—can be computed efficiently without exceeding memory constraints, while preserving scalability for further data-mining extensions.

## Contributions and Broader Impact

The project reframes platform moderation as a measurable system of collective judgment rather than a black-box process of algorithmic enforcement. In recent years, major social platforms have begun to present crowd-based moderation systems as democratic and transparent alternatives to top-down fact-checking. Yet little empirical work has tested whether those systems actually function as examples of collective intelligence—or whether they merely borrow the language of crowds to legitimize their decisions. This project treats that gap as both a methodological and normative problem: if platforms claim to be governed by “the crowd,” then their structures should be measurable by the same principles that make collective judgment reliable.

By operationalizing structural accountability through collective intelligence diagnostics, the project bridges social theory and data science, showing how governance, transparency, and misinformation control can be studied empirically at scale. The broader contribution is not only conceptual but procedural: a framework for auditing human-in-the-loop moderation systems through the traces they leave behind. In quantifying when and how a platform’s “crowd” behaves like a collective intelligence system—and when it doesn’t—this research makes the rhetoric of participatory governance testable. It contributes to computational social science by introducing a scalable diagnostic model for crowd decision-making, and to platform governance studies by showing how public reasoning can be made visible, comparable, and ultimately,accountable.

### Bibliography

Almaatouq, Abdullah, Alejandro Noriega-Campero, Abdulrahman Alotaibi, P. M. Krafft, Mehdi Moussaid, and Alex Pentland. "Adaptive social networks promote the wisdom of crowds." Proceedings of the National Academy of Sciences 117, no. 21 (2020): 11379-11386.

Chuai, Yuwei, Moritz Pilarski, Thomas Renault, David Restrepo-Amariles, Aurore Troussel-Clément, Gabriele Lenzini, and Nicolas Pröllochs. "Community-based fact-checking reduces the spread of misleading posts on social media." arXiv preprint arXiv:2409.08781 (2024).

Drolsbach, Chiara Patricia, Kirill Solovev, and Nicolas Pröllochs. "Community notes increase trust in fact-checking on social media." PNAS nexus 3, no. 7 (2024): pgae217.

Gillespie, Tarleton. Custodians of the Internet: Platforms, content moderation, and the hidden decisions that shape social media. Yale University Press, 2018.

Kim, Junsol, Zhao Wang, Haohan Shi, Hsin-Keng Ling, and James Evans. "Differential impact from individual versus collective misinformation tagging on the diversity of Twitter (X) information engagement and mobility." Nature Communications 16, no. 1 (2025): 973.

Li, Haiwen, Soham De, Manon Revel, Andreas Haupt, Brad Miller, Keith Coleman, Jay Baxter, Martin Saveski, and Michiel A. Bakker. "Scaling Human Judgment in Community Notes with LLMs." arXiv preprint arXiv:2506.24118 (2025).

Lorenz, Jan, Heiko Rauhut, Frank Schweitzer, and Dirk Helbing. "How social influence can undermine the wisdom of crowd effect." Proceedings of the national academy of sciences 108, no. 22 (2011): 9020-9025.

Martel, Cameron, Jennifer Allen, Gordon Pennycook, and David G. Rand. "Crowds can effectively identify misinformation at scale." Perspectives on Psychological Science 19, no. 2 (2024): 477-488.

Matias, J. Nathan. The Civic Labor of Volunteer Moderators Online. Social Media+ Society 5, 2 (April 2019), 2056305119836778. 2019.

Pilarski, Moritz, Kirill Olegovich Solovev, and Nicolas Pröllochs. "Community notes vs. snoping: how the crowd selects fact-checking targets on social media." In Proceedings of the International AAAI Conference on Web and Social Media, vol. 18, pp. 1262-1275. 2024.

Surowiecki, James. "The wisdom of crowds/james surowiecki." NY.: Anchor 5 (2005).